### ISSUE TITLE
Recording always gets trimmed to 4 beats

### ISSUE THREAD

**Issue by deviant-syndrome** [monocasual/giada#613]
**Environment**
- OS: MacOS Catalina 10.15.7
- Giada version: 0.23.0

**Describe the bug**
I cannot record sound for longer than initial sequence size (4beats) at a given tempo. Loop length seems to be correct (same as sequence length), but recorded sound always gets trimmed after first 4 beats

**To Reproduce**
1. Start application
2. Add new sample channel A
3. Load a sample into channel A
4. Set new tempo
5. Put channel A into loop mode 
6. Set the sequence length to 1 bar of 16 beats
7. Add new sample channel B
8. Arm channel B for recording
9. Start recording
10. Stop recording 
11. Put channel B into basic loop mode
12. Disarm channel B recording
13. The recorded sound gets trimmed after 4 beats

**Expected behavior**
I would like the loop sample to play the recorded sound, which duration corresponds to the length of the sequence (e.g. 16 beas of 1 bar)

**Technical information**
The offending lines seem to be [here](https://github.com/monocasual/giada/blob/master/src/core/mixer.cpp#L266-L270): 
```cpp
	const Frame framesToCopy = -1; // copy everything
	const Frame srcOffset    = 0;
	const Frame destOffset   = inputTracker % maxFrames; // loop over at maxFrames

	recBuf.sum(inBuf, framesToCopy, srcOffset, destOffset, inVol);
```
Seems like `maxFrames` is passed or set incorrectly somewhere up the stack. If I remove wrapping with `%` it works. 

Further exploration, led me to conclusion, that this happens, because, changing the number of beats/bars does not call  [Engine::updateMixerModel()](https://github.com/monocasual/giada/blob/master/src/core/engine.cpp#L152), which updates `maxFramesToRec`.

I'll try to create a pull request soon.

**Comment by gvnnz** [comment]
Hi @deviant-syndrome , thanks for the report and sorry for the annoying delay. I'll look into it very soon. Feel free to go ahead with your PR in the meantime.

**Comment by gvnnz** [comment]
Fixed in [667482d](https://github.com/monocasual/giada/commit/667482dde826fba410cc724dc94adc0d40676e58). @deviant-syndrome feel free re-open it otherwise.
